plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -246,26 +243,16 @@ private static BackpressureRestHighLevelClient createClient(
                 clientBuilder.setDefaultCredentialsProvider(credentials);
             });
 
-            awsSecurityConfig.ifPresent(securityConfig -> clientBuilder.addInterceptorLast(new AwsRequestSigner(
-                    securityConfig.getRegion(),
-                    getAwsCredentialsProvider(securityConfig))));
+            awsSignerConfig
+                    .map(awsConfig -> new AwsRequestSigner(awsConfig.getAwsRegion(), awsConfig.getCredentialsProvider()))
+                    .ifPresent(clientBuilder::addInterceptorLast);
```

Comment:
Instead of having optional binder for `AwsSignerCredentialsProvider`. Would it be possible to have a multi set binder for `HttpRequestInterceptor` so we could just call `addInterceptorLast()` for each of them in order they are binded in guice? Then you would provide such in `ElasticsearchAwsModule`.

Reply:
@kokosing This is close to what I originally had, although I applied [this comment](https://github.com/trinodb/trino/pull/8603#discussion_r674098907) from Martin because we want to keep the fact that `ElasticsearchClient` uses the Apache Http Client as an implementation detail and not expose it externally.